File: backend/Services/Parsers/SedecordleSaviorParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class SedecordleSaviorParser : BasicScoreResultParser
    {
        private ILogger<SedecordleSaviorParser> _logger;

        public SedecordleSaviorParser(ILogger<SedecordleSaviorParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override bool Default => false;
        public override string GameName => "Sedecordle Savior";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($@"Daily Savior #\d+\nGuesses: (?<{ScoreGroup}>(\d\d?)|X).*?{ExtraContent}", RegexOptions.Singleline);
        protected override string? ExtraContent => "https://sedecordle.com\n#sedecordle";
        public override string Url => "https://www.sedecordle.com/savior";

        public override string? GetScoreValue(DailyResult dailyResult)
        {
            if (dailyResult.Scores == null)
            {
                return null;
            }

            return "\"" + string.Join(",", Enumerable.Range(0, 16).ToArray()
                .Select((i) => {
                    if (dailyResult.Scores.Count <= i)
                    {
                        return "X";
                    }

                    return dailyResult.Scores[i].ToString();
                })) + "\"";
        }
    }
}

File: lambdas/src/parsers/sedecordlesavior-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';

export class SedecordleSaviorParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Sedecordle Savior';
  readonly helpText = null;
  protected readonly extraContent = 'https://sedecordle.com/\n#sedecordle';
  readonly url = 'https://www.sedecordle.com/savior';
  
  // Matches "Daily Savior #1368\nGuesses: X/21" or "19/21"
  // C# regex: `Daily Savior #\d+\nGuesses: (?<score>(\d\d?)|X).*?{ExtraContent}` with Singleline flag
  protected readonly parser = new RegExp(
    `Daily Savior #\\d+\\nGuesses: (?<${BasicScoreResultParser.SCORE_GROUP}>(\\d\\d?)|X).*?https://sedecordle.com/\\n#sedecordle`,
    's'
  );
  
  getScoreValue(dailyResult: DailyResult): string | null {
    if (!dailyResult.scores) return null;
    
    const values = Array.from({ length: 16 }, (_, i) => {
      if (dailyResult.scores && dailyResult.scores.length > i) {
        return dailyResult.scores[i].toString();
      }
      return 'X';
    });
    
    return `"${values.join(',')}"`;
  }
}
